src/main/java/com/minecolonies/coremod/entity/ai/mobs/util/MobEventsUtils.java

```diff
@@ -0,0 +1,242 @@
+     */
+    private static BlockPos calculateSpawnLocation(final World world, final Colony colony)
+    {
+        final ColonyView colonyView = ColonyManager.getClosestColonyView(world, colony.getCenter());
+        if (colonyView == null)
+        {
+            return null;
+        }
+        final BlockPos center = colonyView.getCenter();
+        final int radius = Configurations.gameplay.workingRangeTownHall;
+
+        final int randomDegree = world.rand.nextInt((int) WHOLE_CIRCLE);
+
+        final double rads = (double) randomDegree / HALF_A_CIRCLE * Math.PI;
+        final double x = Math.round(center.getX() + radius * Math.sin(rads));
+        final double z = Math.round(center.getZ() + radius * Math.cos(rads));
+        return world.getTopSolidOrLiquidBlock(new BlockPos(x, center.getY(), z));
+    }
+
+    /**
+     * Takes a colony and spits out that colony's RaidLevel.
+     *
+     * @param colony The colony to use
+     * @return an int describing the raid level
+     */
+    public static int getColonyRaidLevel(final Colony colony)
+    {
+        int levels = 0;
+
+        @NotNull final List<CitizenData> citizensList = new ArrayList<>();
+        citizensList.addAll(colony.getCitizens().values());
+
+        for (@NotNull final CitizenData citizen : citizensList)
+        {
+            if (citizen.getJob() != null && citizen.getWorkBuilding() != null)
+            {
+                final int buildingLevel = citizen.getWorkBuilding().getBuildingLevel();
+                levels += buildingLevel;
+            }
+        }
+
+        if (colony.getTownHall() != null)
+        {
+            return (levels + colony.getTownHall().getBuildingLevel());
+        }
+        else
+        {
+            return 0;
+        }
+    }
+
+    /**
+     * Reduces barbarian spawns to less than the maximum allowed (set via the config)
+     *
+     * @param total    The current horde size
+     * @param numberOf The number of barbarians which we are reducing
+     * @return the new number that the barbarians should be set to.
+     */
+    private static int equalizeBarbarianSpawns(final int total, final int numberOf)
+    {
+        int returnValue = numberOf;
+        if (total > MAX_SIZE)
+        {
+            returnValue = total - MAX_SIZE;
+
+            if (returnValue < 0)
+            {
+                return 0;
+            }
+            return returnValue;
+        }
+        return returnValue;
+    }
+
+    public static boolean isItTimeToRaid(final World world, final Colony colony)
+    {
+        if (colony.getCitizens().size() < NUMBER_OF_CITIZENS_NEEDED)
+        {
+            //System.out.println("Debug1");
+            return false;
+        }
+        if ((world.getWorldTime() - TICKS_AFTER_HALF_DAY) % HALF_MINECRAFT_DAY == 0)
+        {
+            System.out.println("Debug2");
+            if (world.isDaytime())
+            {
+                System.out.println("Debug3");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS106)